File: src/test/java/com/lkimilhol/matchingproject/member/domain/NicknameTest.java
package com.lkimilhol.matchingproject.member.domain;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class NicknameTest {

    @DisplayName("생성")
    @Test
    void create() {
        // given
        Nickname nickname = new Nickname("김메이어");
        // when
        // then
        assertThat(nickname).isNotNull();
    }
}

File: src/test/kotlin/com/lkimilhol/matchingproject/member/domain/NicknameTest.kt
package com.lkimilhol.matchingproject.member.domain

import com.lkimilhol.matchingproject.exception.EmptyException
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class NicknameTest {
    @Test
    fun `이름은 빈값 일 수 없다`() {
        // given
        // when
        // then
        Assertions.assertThatThrownBy { Nickname("") }
            .isInstanceOf(EmptyException::class.java)
    }

    @Test
    fun `생성`() {
        // given
        val nickname = "테스트"
        // when
        val nicknameNew = Nickname(nickname)
        // then
        assertThat(nicknameNew).isEqualTo(Nickname(nickname))
    }
}